### PR TITLE
Add external mysql option for pipeline and metadata

### LINKED ISSUE #905: Support external mysql database for metadata and pipeline
Currently, we use PersistVolume based mysql database for these projects and upgrade and performance becomes a challenges. 

One solution is to use external mysql database, we can add overlays and make this as another option for users. 

Address https://github.com/kubeflow/kubeflow/issues/4546

### PR DESCRIPTION
**Pull request by Jeffwan**
**Which issue is resolved by this Pull Request:**
Resolves #905 

**Description of your changes:**
Add two overlays to allow user to use external mysql servers. 

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate-changed-only`
    3. `make test`

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/906)

### CODE REVIEW

**Comment by Jeffwan** [comment]
```
INFO|2020-02-12T20:36:07|/src/kubeflow/testing/py/kubeflow/testing/argo_client.py|31| Workflow kubeflow-manifests-presubmit-kfctl-go-iap-906-18ef263-7328-2a54 in namespace kubeflow-test-infra; phase=Failed
```

Seems it's unrelated and this PR only add new overlays.

File: pipeline/api-service/overlays/external-mysql/params.env [hunk 1]
```diff
@@ -0,0 +1,3 @@
+mysqlHost=
+mysqlUser=
```
Comment (Jeffwan) [inline pipeline/api-service/overlays/external-mysql/params.env]: Temporarily put username  and password to configmap. These will be used to replace placeholder in pipeline configmap. The challenge is the value from secret is encoded string and I can not find way to decode it easily.
secretGenerator can not use `StringData` and upstream issue is tracked here.  https://github.com/kubernetes-sigs/kustomize/issues/1444

**Comment by zhenghuiwang** [comment]
/lgtm

The metadata part looks good. Do you plan to make it the default overlay for aws?

**Comment by Jeffwan** [comment]
@zhenghuiwang 
No, this is an option for users want to use cloud managed SQL service.

**Comment by xaniasd** [comment]
@Jeffwan thanks for this PR. How could I customize the values in secrets.env?

**Comment by Jeffwan** [comment]
@xaniasd Customize the values in secrets.env?  Do you mean how to apply overlays? You can do like this for metadata and pipeline. Make sure you fill secrets.env and params.env

image

**Comment by xaniasd** [comment]
@Jeffwan no my question was about filling in secrets.env. I can customize parameters in params.env with the parameters map like so:
```yaml
- kustomizeConfig:
    overlays:
    - application
    parameters:
    - name: MY_HOST
      value: db
```
As far as I can see in kfctl code, this is not possible for secrets.env files (maybe for good reason too). So how can I change the password?

**Comment by Jeffwan** [comment]
@xaniasd Hi, https://github.com/kubeflow/kfctl/blob/master/pkg/apis/apps/kfdef/v1/application_types.go doesn't have field to set secrets. Currently, only few application use secrets and I think it could be a feature request. You can `kfctl build` and manually update `secrets.env` at this moment.

**Comment by thesuperzapper** [comment]
@Jeffwan do you have plans to add support for external DB's in the `katib-controller` manifests?

See here:
https://github.com/kubeflow/manifests/blob/v1.0.2/katib/katib-controller/base/katib-mysql-deployment.yaml

**Comment by Jeffwan** [comment]
@thesuperzapper 

There's some ongoing discussion there. Katib maintainers has some concerns on the base manifests since there's standalone users. But I think that's the trend, we should have similar pattern there. I will check it later. 

for the RDS documentation, please check this website after this PR merged 

https://github.com/kubeflow/website/pull/1934

**Pull request merged**